library/ssl_cli.c

```diff
@@ -1453,16 +1453,16 @@ static int ssl_parse_cid_ext( mbedtls_ssl_context *ssl,
     {
         MBEDTLS_SSL_DEBUG_MSG( 1, ( "CID extension unexpected" ) );
         mbedtls_ssl_send_alert_message( ssl, MBEDTLS_SSL_ALERT_LEVEL_FATAL,
-                                     MBEDTLS_SSL_ALERT_MSG_HANDSHAKE_FAILURE );
-        return( MBEDTLS_ERR_SSL_BAD_HS_SERVER_HELLO );
+                                        MBEDTLS_SSL_ALERT_MSG_ILLEGAL_PARAMETER );
```

Comment:
Note that ssl_parse_encrypt_then_mac_ext alerts/returns a DECODE_ERROR in ssl_srv.c, but UNSUPPORTED_EXT/ILLEGAL_PARAMETER in ssl_cli.c - should these be consistent (is the former incorect)?

Reply:
`unsupported_extension` is only ever sent by clients (see [RFC 5246](https://datatracker.ietf.org/doc/html/rfc5246#section-7.2.2)), when the server uses the extension even though the client has not declared support for it. The opposite case doesn't happen since the client declares its extensions first.